pyclam/manifold.py

```diff
@@ -232,14 +234,52 @@ def local_fractal_dimension(self) -> float:
         if 'local_fractal_dimension' not in self.cache:
             logging.debug(f'building cache for {self}')
             if self.nsamples == 1:
-                return 0.
-            count = [d <= (self.radius / 2)
-                     for batch in iter(self)
-                     for d in self.distance_from(batch)]
-            count = np.sum(count)
-            self.cache['local_fractal_dimension'] = count if count == 0. else np.log2(len(self.argpoints) / count)
+                self.cache['local_fractal_dimension'] = 1.
+            else:
+                count = sum([
+                    1 if distance <= (self.radius / 2) else 0
+                    for batch in iter(self)
+                    for distance in self.distance_from(batch)
+                ])
+                self.cache['local_fractal_dimension'] = 1. if count == 0 else np.log2(self.cardinality / count)
         return self.cache['local_fractal_dimension']
```

Comment:
When can we drop support for 3.6 and just use `functools.cached_property`

Reply:
When we switch to a Rust backend, we will use proper caching. For now, I don't really want to drop python 3.6